pkg/security/security_profile/manager.go

```diff
@@ -0,0 +1,2376 @@
+		defer adTagsTickerChanTimer.Stop()
+		adTagsTickerChan = adTagsTickerChanTimer.C
+
+		adLoadControlTicker := time.NewTicker(m.config.RuntimeSecurity.ActivityDumpLoadControlPeriod)
+		defer adLoadControlTicker.Stop()
+		adLoadControlTickerChan = adLoadControlTicker.C
+	} else {
+		adCleanupTickerChan = make(chan time.Time)
+		adTagsTickerChan = make(chan time.Time)
+		adLoadControlTickerChan = make(chan time.Time)
+	}
+
+	if m.config.RuntimeSecurity.ActivityDumpEnabled && m.config.RuntimeSecurity.SecurityProfileEnabled {
+		silentWorkloadsTicker := time.NewTicker(m.config.RuntimeSecurity.ActivityDumpSilentWorkloadsTicker)
+		defer silentWorkloadsTicker.Stop()
+		silentWorkloadsTickerChan = silentWorkloadsTicker.C
+	} else {
+		silentWorkloadsTickerChan = make(chan time.Time)
+	}
+
+	if m.config.RuntimeSecurity.SecurityProfileEnabled {
+		_ = m.resolvers.TagsResolver.RegisterListener(tags.WorkloadSelectorResolved, m.onWorkloadSelectorResolvedEvent)
+		_ = m.resolvers.TagsResolver.RegisterListener(tags.WorkloadSelectorDeleted, m.onWorkloadDeletedEvent)
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	seclog.Infof("security profile manager started")
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-adCleanupTickerChan:
+			m.cleanup()
+		case <-adTagsTickerChan:
+			m.resolveTagsAll()
+		case <-adLoadControlTickerChan:
+			m.triggerLoadController()
+		case ad := <-m.snapshotQueue:
+			if err := m.snapshot(ad); err != nil {
+				seclog.Errorf("couldn't snapshot [%s]: %v", ad.Profile.Metadata.ContainerID, err)
+			}
+		case <-silentWorkloadsTickerChan:
+			m.handleSilentWorkloads()
+		case newProfile := <-m.newProfiles:
+			m.onNewProfile(newProfile)
+		}
+	}
+}
+
+// getExpiredDumps returns the list of dumps that have timed out and remove them from the active dumps
+func (m *Manager) getExpiredDumps() []*dump.ActivityDump {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	var expiredDumps []*dump.ActivityDump
+	var newDumps []*dump.ActivityDump
+	for _, ad := range m.activeDumps {
+		if time.Now().After(ad.Profile.Metadata.End) || ad.GetState() == dump.Stopped {
+			expiredDumps = append(expiredDumps, ad)
+			delete(m.ignoreFromSnapshot, ad.Profile.Metadata.CGroupContext.CGroupFile)
+		} else {
+			newDumps = append(newDumps, ad)
+		}
+	}
+
+	m.activeDumps = newDumps
+	return expiredDumps
+}
+
+// cleanup
+func (m *Manager) cleanup() {
+	// fetch expired dumps
+	dumps := m.getExpiredDumps()
+
+	for _, ad := range dumps {
+		m.finalizeKernelEventCollection(ad, true)
+		seclog.Infof("tracing stopped for [%s]", ad.GetSelectorStr())
+
+		// persist dump if not empty
+		if !ad.Profile.IsEmpty() && ad.Profile.GetWorkloadSelector() != nil {
+			if err := m.persist(ad.Profile, m.configuredStorageRequests); err != nil {
+				seclog.Errorf("couldn't persist dump [%s]: %v", ad.GetSelectorStr(), err)
+			} else if m.config.RuntimeSecurity.SecurityProfileEnabled {
```

Comment:
should we remove the `else` here?

Reply:
I think we need it in case ActivityDumps are enabled but SecurityProfiles aren't, that way we avoid pushing the new AD to the newProfile channel (which is never read when SecurityProfiles aren't enabled)
I added a comment regarding this cdf65213d45aec3b865990da8f9503790a62038a